lib/public/components/qcFlags/qcFlagsBreadcrumbs.js

```diff
@@ -0,0 +1,60 @@
+import { h, iconWarning } from '/js/src/index.js';
+import { breadcrumbs } from '../common/navigation/breadcrumbs.js';
+import { tooltip } from '../common/popover/tooltip.js';
+import { frontLink } from '../common/navigation/frontLink.js';
+import { RunQualities } from '../../domain/enums/RunQualities.js';
+import spinner from '../common/spinner.js';
+import { BAD_RUN_IN_ASYNC_QC_MESSAGE } from './badRunInAsyncQcMessage.js';
+
+/**
+ * Render QC flag(-s) breadcrumbs
+ *
+ * @param {object} qcBreacrumbsElements breadcrumbs elements
+ * @param {Run} remoteRun run remote data
+ * @param {number} [remoteDataPass] data pass remote data -- exclusive with `remoteSimulationPass`
+ * @param {object} [remoteSimulationPass] simulation pass remote data -- exclusive with `remotedataPass`
+ * @param {number} remoteDplDetector dpl detector remote data
+ * @return {Component} breadcrumbs
+ */
+export const qcFlagsBreadcrumbs = ({ remoteDataPass, remoteSimulationPass, remoteRun, remoteDplDetector }) => {
+    if (!remoteDataPass && !remoteSimulationPass) {
+        throw new Error('`remoteDataPass` or `remoteSimulationPass` is required');
+    }
+    if (remoteDataPass && remoteSimulationPass) {
+        throw new Error('`remoteDataPass` and `remoteSimulationPass` are exclusive options');
+    }
+
+    return breadcrumbs([
+        h('h2', 'QC'),
+        remoteDataPass ?
+            remoteDataPass.match({
+                Success: ({ id, name }) => h('h2', frontLink(name, 'runs-per-data-pass', { dataPassId: id })),
+                Failure: () => tooltip(h('.f3', iconWarning()), 'Not able to load data pass info'),
+                Loading: () => h('', spinner({ size: 2, absolute: false })),
+                NotAsked: () => tooltip(h('.f3', iconWarning()), 'No data pass data was asked for'),
+            })
+            : remoteSimulationPass.match({
+                Success: ({ id, name }) => h('h2', frontLink(name, 'runs-per-simulation-pass', { simulationPassId: id })),
+                Failure: () => tooltip(h('.f3', iconWarning()), 'Not able to load simulation pass info'),
+                Loading: () => h('', spinner({ size: 2, absolute: false })),
+                NotAsked: () => tooltip(h('.f3', iconWarning()), 'No simulation pass data was asked for'),
+            }),
+        remoteRun.match({
+            Success: ({ runNumber, runQuality }) => runQuality === RunQualities.BAD
+                ? tooltip(
+                    h('h2.danger', frontLink(runNumber, 'run-detail', { runNumber })),
+                    BAD_RUN_IN_ASYNC_QC_MESSAGE,
```

Comment:
I don't understand the need of the message here

Reply:
no need, but it's clearer for user when run is marked out and why it's marked out is described in tooltip

Reply:
I think it's not, braedcrumbs are simply navigation links and this add too much things to think about

Reply:
ok, tooltip removed